styx-common/src/main/java/com/spotify/styx/model/data/WFIExecutionBuilder.java

```diff
@@ -221,6 +220,12 @@ public Void timeout(WorkflowInstance workflowInstance) {
 
       executionStatusList.add(ExecStatus.create(eventTs, "TIMEOUT", Optional.empty()));
 
+      // we might get timeout before triggerExecution, and in that case we take best effort to
+      // set trigger timestamp
+      if (triggerTs == null) {
```

Comment:
How about moving this check at the beginning of the method and also return if this if condition is true (after setting `triggerTS`)? So we avoid showing weird executions

Reply:
Good point! Let's try.